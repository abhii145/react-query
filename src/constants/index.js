
export const SORT_OPTIONS = {
  "Price: Low to High": { sortBy: "price", order: "asc" },
  "Price: High to Low": { sortBy: "price", order: "desc" },
  "Newest First": { sortBy: "date", order: "desc" },
  "Highest Rating": { sortBy: "rating", order: "desc" },
  feature: { sortBy: "feature", order: "asc" },
}

export const DEFAULT_SEARCH_PARAMS = {
  skip: 0,
  limit: 10,
  search: "",
  category: "All",
  sortBy: "price",
  order: "asc",
}