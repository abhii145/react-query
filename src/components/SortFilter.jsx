import { SORT_OPTIONS } from "../constants"
import PropTypes from "prop-types"

const SortFilter = ({ onSortChange, selectedSort }) => (
  <div className="flex items-center gap-2">
    <label htmlFor="sort" className="text-sm font-medium text-gray-600">
      Sort By:
    </label>
    <select
      id="sort"
      onChange={onSortChange}
      value={selectedSort}
      className="p-2 border border-gray-300 rounded-md text-sm focus:ring-2 focus:ring-blue-500 focus:outline-none"
    >
      {Object.keys(SORT_OPTIONS).map((option) => (
        <option key={option} value={option}>
          {option}
        </option>
      ))}
    </select>
  </div>
)
SortFilter.propTypes = {
  onSortChange: PropTypes.func.isRequired,
  selectedSort: PropTypes.string.isRequired,
}

export default SortFilter
