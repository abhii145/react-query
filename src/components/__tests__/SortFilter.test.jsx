import { render, screen, fireEvent } from "@testing-library/react"
import SortFilter from "../SortFilter"
import { SORT_OPTIONS } from "../../constants"
import { it, expect, describe, vi } from "vitest"

describe("SortFilter", () => {
  const mockOnSortChange = vi.fn()
  const selectedSort = "Price: Low to High"

  it("renders correctly", () => {
    render(
      <SortFilter onSortChange={mockOnSortChange} selectedSort={selectedSort} />
    )
    expect(screen.getByLabelText(/sort by/i)).toBeInTheDocument()
    Object.keys(SORT_OPTIONS).forEach((option) => {
      expect(screen.getByText(option)).toBeInTheDocument()
    })
  })

  it("renders without crashing", () => {
    render(
      <SortFilter onSortChange={mockOnSortChange} selectedSort={selectedSort} />
    )
    expect(screen.getByLabelText(/sort by/i)).toBeInTheDocument()
  })

  it("correct option is selected", () => {
    render(
      <SortFilter onSortChange={mockOnSortChange} selectedSort={selectedSort} />
    )
    expect(screen.getByDisplayValue(selectedSort)).toBeInTheDocument()
  })

  it("calls onSortChange when an option is selected", () => {
    render(
      <SortFilter onSortChange={mockOnSortChange} selectedSort={selectedSort} />
    )
    fireEvent.change(screen.getByLabelText(/sort by/i), {
      target: { value: "Price: High to Low" },
    })
    expect(mockOnSortChange).toHaveBeenCalled()
  })
})
