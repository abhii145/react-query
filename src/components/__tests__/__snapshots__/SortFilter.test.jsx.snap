// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SortFilter > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center gap-2"
  >
    <label
      class="text-sm font-medium text-gray-600"
      for="sort"
    >
      Sort By:
    </label>
    <select
      class="p-2 border border-gray-300 rounded-md text-sm focus:ring-2 focus:ring-blue-500 focus:outline-none"
      id="sort"
    >
      <option
        value="feature"
      >
        feature
      </option>
      <option
        value="Price: Low to High"
      >
        Price: Low to High
      </option>
      <option
        value="Price: High to Low"
      >
        Price: High to Low
      </option>
      <option
        value="Newest First"
      >
        Newest First
      </option>
      <option
        value="Highest Rating"
      >
        Highest Rating
      </option>
    </select>
  </div>
</DocumentFragment>
`;
